c10/core/impl/alloc_cpu.h

```diff
@@ -4,8 +4,21 @@
 
 #include <cstddef>
 
+#ifdef __linux__
+#include <sys/mman.h>
+#include <cstdlib>
+#endif
+
 namespace c10 {
 
+#ifdef __linux__
+// since the default thp pagesize is 2MB, enable thp only
+// for buffers of size 2MB or larger to avoid memory bloating
+constexpr size_t gAlloc_threshold_thp = 2 * 1024 * 1024;
+static const char* thp_env = std::getenv("C10_THP_MEM_ALLOC_ENABLE");
+const bool gIs_c10_thp_mem_alloc_enabled = std::atoi(thp_env != nullptr ? thp_env : "0");
```

Comment:
This initializer is called at library load time, but if it were to be implemented as an inline function in alloc_cpu.cpp it could be postponed until the first use time

Something like(also, c10 name is redundant as it is already in c10 namespace):
```c++
const bool is_thp_alloc_enabled() {
  static bool value = [] {
      auto thp_env = std::getenv("C10_THP_MEM_ALLOC_ENABLE");
      return std::atoi(thp_env != nullptr ? thp_env : "0");
  };
  return value;
}

Reply:
My intention was to avoid reading the env for every alloca call.